l10n_ch_pain_base/models/account_move_line.py

```diff
@@ -0,0 +1,17 @@
+# copyright 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _prepare_payment_line_vals(self, payment_order):
+        vals = super()._prepare_payment_line_vals(payment_order)
+        if self.move_id and self.move_id._is_isr_ref():
```

Comment:
`is_isr_ref` is not enough here. It only checks that we have a 27 digits reference.

What you are looking for is to know if the tuple (bank_account, ref) is of ISR type.

The bank account must be an ISR issuer. Postal account number format starting by `01` or `03`
https://github.com/odoo/odoo/blob/13.0/addons/l10n_ch/models/res_bank.py#L52

Thus you want something like this:

```suggestion
        if self.move_id.partner_bank_id._is_isr_issuer() and self.move_id._is_isr_ref():
```

Reply:
@yvaucher ... doing a functional revision I get next error:

![image](https://user-images.githubusercontent.com/8027645/94674455-7a3f4f80-0318-11eb-9597-feeacb111426.png)

so a new commit with the fix will be included